java/connector-node/risingwave-connector-service/src/main/java/com/risingwave/connector/source/core/DbzChangeEventConsumer.java

```diff
@@ -168,14 +181,66 @@ var record = event.value();
                                         .setSourceTsMs(trxTs)
                                         .build();
                         LOG.debug("transaction => {}", message);
                         respBuilder.addEvents(message);
                         break;
                     }
+
+                case SCHEMA_CHANGE:
+                    {
+                        var ddl = ((Struct) record.value()).getString("ddl");
+                        if (ddl.contains("CREATE") || ddl.contains("DROP")) {
+                            LOG.info("skip create/drop table event");
+                            continue;
+                        }
+
+                        var sourceStruct = ((Struct) record.value()).getStruct("source");
+                        if (sourceStruct == null) {
+                            throw new CdcConnectorException(
+                                    "source field is missing in schema change event");
+                        }
+
+                        // upstream event time
+                        long sourceTsMs = sourceStruct.getInt64("ts_ms");
+                        // concat full table name, right now we only support MySQL schema change
+                        // event
+                        var fullTableName =
+                                String.format(
+                                        "%s.%s",
+                                        sourceStruct.getString("db"),
+                                        sourceStruct.getString("table"));
+                        byte[] payload =
+                                payloadConverter.fromConnectData(
+                                        record.topic(), record.valueSchema(), record.value());
+
+                        var message =
+                                msgBuilder
+                                        .setMsgType(CdcMessage.CdcMessageType.SCHEMA_CHANGE)
+                                        .setFullTableName(fullTableName)
+                                        .setPayload(new String(payload, StandardCharsets.UTF_8))
+                                        .setSourceTsMs(sourceTsMs)
+                                        .build();
+                        LOG.debug(
+                                "offset => {}, key => {}, payload => {}",
+                                message.getOffset(),
+                                message.getKey(),
+                                message.getPayload());
+                        respBuilder.addEvents(message);
+
+                        // emit the schema change event as a single response
+                        respBuilder.setSourceId(sourceId);
+                        var response = respBuilder.build();
+                        outputChannel.put(response);
+
+                        // reset the response builder
+                        respBuilder = GetEventStreamResponse.newBuilder();
+                        break;
+                    }
```

Comment:
Based on dbz's [spec](https://debezium.io/documentation/reference/2.6/connectors/mysql.html#mysql-schema-change-topic):
> The ddl field can contain multiple DDL statements.

> tableChanges | An array of one or more items that contain the schema changes generated by a DDL command.

it indicates that there can be multiple DDL statements in one schema change event, possibly corresponding to multiple tables in the same DB in one schema change event. However, the implementation here assumes that there is only one DDL statement corresponding to one table in a schema change event.

Reply:
> However, the implementation here assumes that there is only one DDL statement corresponding to one table in a schema change event.

This is not true. Here we just deliberately let each schema event in a separate message batch, so that data messages and schema messages would not stay in the same batch.

Reply:
I mean based on the dbz's spec, one schema change message can contain multiple DDL statements.

Reply:
>However, the implementation here assumes that there is only one DDL statement corresponding to one table in a schema change event.

I think it is the extraction of `fullTableName` gives you this impression, and I confirmed that the full table name is useless for schema change message, so we can remove it.
And in [Meta](https://github.com/risingwavelabs/risingwave/pull/17876/files#diff-6d6a3906f0c7c782273f1310c8497646663a548615d2f2afaf9fe7449cb63187R950-R956) we process each `TableSchemaChange` to do the schema change, I think we can handle the case that _"The ddl field can contain multiple DDL statements."_